core/src/test/java/org/pentaho/di/connections/vfs/provider/ConnectionFileNameParserTest.java

```diff
@@ -270,4 +271,48 @@ protected String shuffleString( String characterSet ) {
     return sb.toString();
   }
   // endregion
+
+  // region isValidConnectionNameCharacter
+  @Test
+  public void testIsValidConnectionNameCharacterReturnsFalseForInvalidCharacters() {
+    for ( char c : CONNECTION_NAME_INVALID_CHARACTERS.toCharArray() ) {
+      assertFalse( fileNameParser.isValidConnectionNameCharacter( c ) );
+    }
+  }
+
+  @Test
+  public void testIsValidConnectionNameCharacterReturnsTrueOnValidCharacters() throws FileSystemException {
```

Comment:
you have a code smell here: https://sonar.orl.eng.hitachivantara.com/project/issues?pullRequest=9455&resolved=false&types=CODE_SMELL&open=AZBzRVUFpCtKek2RFu0W&id=org.pentaho.di%3Apdi

Reply:
Nice catch :-)